ServerCore/Pages/Teams/Details.cshtml.cs

```diff
@@ -1,328 +1,331 @@
+            if (Team == null)
+            {
+                return NotFound("No team found with id '" + teamId + "'.");
+            }
+
+            // Existing team members
+            Members = await _context.TeamMembers.Where(members => members.Team.ID == Team.ID).ToListAsync();
+            StringBuilder emailList = new StringBuilder("");
+            foreach (TeamMembers Member in Members)
+            {
+                emailList.Append(Member.Member.Email + "; ");
+            }
+            Emails = emailList.ToString();
+
+            // Team applicants
+            Users = await (from application in _context.TeamApplications
+                           where application.Team == Team &&
+                           !((from teamMember in _context.TeamMembers
+                              where teamMember.Member == application.Player &&
+                         teamMember.Team.Event == Event
+                              select teamMember).Any())
+                           select new Tuple<PuzzleUser, int>(application.Player, application.ID)).ToListAsync();
+
+            // Lunch
+            if (Event.EventHasTeamSwag)
+            {
+                PlayersPerLunch = 1;
+                if (Event.PlayersPerLunch != null)
+                {
+                    PlayersPerLunch = Event.PlayersPerLunch.Value;
+                }
+
+                GlobalMaxLunches = (int)Math.Ceiling((double)Event.MaxTeamSize / (double)PlayersPerLunch);
+                int totalMembers = Members.Count;
+                int remoteMembers = await (from player in _context.PlayerInEvent
+                                           join member in _context.TeamMembers on player.PlayerId equals member.Member.ID
+                                           where member.Team.ID == Team.ID &&
+                                           player.IsRemote
+                                           select player).CountAsync();
+                EligibleForLunch = totalMembers - remoteMembers;
+
+                EditableLunches = (int)Math.Ceiling((double)EligibleForLunch / (double)PlayersPerLunch);
+
+                Lunches = await (from lunch in _context.TeamLunch
+                                 where lunch.TeamId == teamId
+                                 orderby lunch.ID
+                                 select lunch).ToListAsync();
+
+                double possibleInPersonMembers = Event.MaxTeamSize - remoteMembers;
+                SoftMaxLunches = (int)Math.Ceiling(possibleInPersonMembers / (double)PlayersPerLunch);
+            }
+            LunchOptions = Event.LunchOptions.Split(";");
+            for (int i = 0; i < LunchOptions.Length; i++) 
+            {
+                LunchOptions[i] = LunchOptions[i].Split(":")[0];
+                LunchOptions[i] = LunchOptions[i].Substring(LunchOptions[i].IndexOf("\"") + 1, LunchOptions[i].LastIndexOf("\"") - LunchOptions[i].IndexOf("\"") - 1);

```

Comment:
So this just throws the details away? If so then the details page should make it clear that the details are for single player events only.